demo-node/entrypoint.bash

```diff
@@ -44,20 +46,21 @@ curl -X POST \
 #  http://127.0.0.1:5160/api/v0/pipeline/create
 
 # !! Not started because the data has already been imported while building the image.
-# Newlines improve readability in the app.
-suricata_pipe="tenzir 'load https https://storage.googleapis.com/tenzir-datasets/M57/suricata.json.zst \n| decompress zstd \n| read suricata \n| where #schema != \\\"suricata.stats\\\" \n| import'"
+suricata_pipe_formatted=$(echo $SURICATA_PIPE | sed -r 's/ \| /\\n| /g' | sed -r 's/"/\\\\\\\"/g' )
+echo $suricata_pipe_formatted
 curl -X POST \
   -H "Content-Type: application/json" \
-  -d "{\"name\": \"M57 Suricata Import\", \"definition\": \"${suricata_pipe}\", \"start_when_created\": false}" \
+  -d "{\"name\": \"M57 Suricata Import\", \"definition\": \"tenzir '${suricata_pipe_formatted}'\", \"start_when_created\": false}" \
   http://127.0.0.1:5160/api/v0/pipeline/create
+echo
 
 # !! Not started because the data has already been imported while building the image.
-# Newlines improve readability in the app.
-zeek_pipe="tenzir 'load https https://storage.googleapis.com/tenzir-datasets/M57/zeek-all.log.zst \n| decompress zstd \n| read zeek-tsv \n| import'"
-
+zeek_pipe_formatted=$(echo $ZEEK_PIPE | sed -r 's/ \| /\\n| /g' | sed -r 's/"/\\\\\\\"/g' )
+echo $suricata_pipe_formatted
 curl -X POST \
   -H "Content-Type: application/json" \
-  -d "{\"name\": \"M57 Zeek Import\", \"definition\": \"${zeek_pipe}\", \"start_when_created\": false}" \
+  -d "{\"name\": \"M57 Zeek Import\", \"definition\": \"tenzir '${zeek_pipe_formatted}'\", \"start_when_created\": false}" \
```

Comment:
Would it make sense to change this to actually start the pipeline, and then to wait until it is completed? That way, the pipeline will show up as completed rather than as created.

Reply:
To me, the very best would even be not to preload the data at build time and let the user see the import of the Zeek and Suricata datasets happening live. This way the demo nodes would connect even faster because they would be more lightweight in every sense. But unfortunately if I remember correctly there were some OOM issues with that.

Reply:
You can probe whether all pipelines are done like this:

```bash
curl -X POST http://127.0.0.1:5160/api/v0/pipeline/list |
  jq -e '.pipelines | map(.state) | all(. == "completed")'
```

Which you can use in a loop with some sleep to wait until both pipelines are done.

Afterwards, you can issue a `tenzir-ctl rebuild --all` to optimize the DB.

Unless I'm missing something, this would be the strictly better option.

> To me, the very best would even be not to preload the data at build time and let the user see the import of the Zeek and Suricata datasets happening live.

I'd also be in favor of that, but it would require expanding the tour a fair bit. It would, however, allow us to remove the special image entirely.